starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/templates/buildGradle.rocker.raw

```diff
@@ -438,6 +438,9 @@ dependencies {
 @if (features.contains("spring")) {
     @dependency.template("org.springframework.boot", "spring-boot-starter", "implementation", null)
 }
+@if (features.contains("spring-tx")) {
+    @dependency.template("io.micronaut.sql", "micronaut-hibernate-jpa-spring", "implementation", null)
```

Comment:
This doesn't make any sense to me. You can use Spring transactions with just JDBC or just Micronaut Data JDBC so it has nothing to do with `micronaut-hibernate-jpa-spring`

If we are going to merge this PR then the feature should also check if Hibernate is selected to apply this and should also work with just JDBC